spec/models/target_spec.rb

```diff
@@ -32,74 +36,95 @@
+
+  describe 'associations' do
+    it { should belong_to(:conference) }
+    it { should belong_to(:campaign) }
+  end
+
   describe '#get_progress' do
     it 'returns zero if there are no registrations' do
-      conference = build(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:registrations])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+      expect(registration_target.get_progress).to eq('0')
     end
 
     it 'returns 10 if there one registrations of 10' do
-      conference = create(:conference)
-      target = create(:target, target_count: 10, unit: Target.units[:registrations])
-      registration = create(:registration)
-
-      conference.targets = [target]
-      conference.registrations = [registration]
+      create(:registration, conference: registration_target.conference)
 
-      expect(target.get_progress).to eq('10')
+      expect(registration_target.get_progress).to eq('10')
     end
 
     it 'returns zero if there are no submissions' do
-      conference = create(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:submissions])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+      expect(submission_target.get_progress).to eq('0')
     end
 
     it 'returns 10 if there one submissions of 10' do
-      conference = create(:conference)
-      target = create(:target, target_count: 10, unit: Target.units[:submissions])
-      event = create(:event)
-
-      conference.targets = [target]
-      conference.program.events = [event]
+      submission_target.conference.program.events = [event]
 
-      expect(target.get_progress).to eq('10')
+      expect(submission_target.get_progress).to eq('10')
     end
 
     it 'returns zero if there are no program minutes' do
-      conference = create(:conference)
-      target = build(:target, target_count: 10, unit: Target.units[:program_minutes])
-      conference.targets = [target]
-
-      expect(target.get_progress).to eq('0')
+      expect(program_minutes_target.get_progress).to eq('0')
     end
 
     it 'returns 10 if there are 30 program minutes of 300' do
-      conference = create(:conference)
-      target = create(:target, target_count: 300, unit: Target.units[:program_minutes])
-      event = create(:event)
+      program_minutes_target.conference.program.events = [event]
 
-      conference.targets = [target]
-      conference.program.events = [event]
+      expect(program_minutes_target.get_progress).to eq('10')
+    end
+  end
+
+  describe '#get_campaign' do
+    let(:campaign_registration_target) { create(:campaign_target, target_count: 10, unit: Target.units[:registrations]) }
+    let(:campaign_submission_target) { create(:campaign_target, target_count: 10, unit: Target.units[:submissions]) }
+    let(:campaign_program_minutes_target) { create(:campaign_target, target_count: 300, unit: Target.units[:program_minutes]) }
 
```

Comment:
Not that this is wrong, but since you are not doing anything different with these variables, you could have simply defined `let(:campaign) { create(:campaign }` in the beginning of the file and then associate campaign with registration_target, submission_target and program_minutes_target, and then you would not need to define more objects here.